src/main/java/org/cyclops/integrateddynamics/core/evaluate/operator/Operators.java

```diff
@@ -1182,6 +1183,24 @@ public IValue evaluate(OperatorBase.SafeVariablesGetter variables) throws Evalua
                 return ValueTypeList.ValueList.ofFactory(new ValueTypeListProxySlice<>(list, from.getRawValue(), to.getRawValue()));
             }).build());
 
+    public static final IOperator LIST_INTERSECTION = REGISTRY.register(OperatorBuilders.LIST
+            .inputTypes(ValueTypes.LIST, ValueTypes.LIST)
+            .renderPattern(IConfigRenderPattern.PREFIX_2_LONG).output(ValueTypes.LIST)
+            .symbol("∩").operatorName("intersection")
+            .function(variables -> {
+                ValueTypeList.ValueList list1 = variables.getValue(0, ValueTypes.LIST);
+                IValueTypeListProxy<IValueType<IValue>, IValue> rawList1 = list1.getRawValue();
+                ValueTypeList.ValueList list2 = variables.getValue(1, ValueTypes.LIST);
+                IValueTypeListProxy<IValueType<IValue>, IValue> rawList2 = list2.getRawValue();
+                Stream<IValue> result = new ArrayList<>(Sets.newLinkedHashSet(rawList1)).stream().filter(value1 -> {
```

Comment:
Why is a set used here? We probably don't need it I guess.

Reply:
Actually good point. I've refacted this to use set methods, which are bound to be more efficient - with a bonus of fixing an issue where and item being present...

That said, it also raises the question of what the expected behaviour would be when multiple item are present in a given list.
![image](https://user-images.githubusercontent.com/286341/195001351-d143fe48-cb28-4d59-9194-2c6df54a41c2.png)
Should that show a single 0 (ie, there are 0s in both lists) or two 0s?

Reply:
The problem is that `IValue` does not implement `hashCode` (AFAIK), so the performance benefit will be minimal.

> That said, it also raises the question of what the expected behaviour would be when multiple item are present in a given list.

Perhaps it's better to leave that behaviour undefined. If needed, the user can still use `uniq`.

Reply:
Does that mean you want me to pull back that last commit, and get rid of the sets?

Reply:
I guess we can keep the new version. But definitely test it as well with more complex datatypes (like recipes), to make sure we don't run into issues due to the lacking `hashCode`.

Reply:
I just tried recipes - seems to work. Making a list of recipes is a pain though :P

There may be performance gains to be had (in this, and perhaps other operators) but it doesn't seem like it's *required* at this stage.